content/en/docs/concepts/security/hardening-guide/scheduler.md

```diff
@@ -0,0 +1,82 @@
+{{</table>}}
+
+### TLS configuration options
+{{<table caption="Address Configurations">}}
+| Configuration | Description | Security Hardening Advice |
+| --- | --- | --- |
+|`requestheader-client-ca-file`|Root certificate bundle to use to verify client certificates on incoming requests before trusting usernames in headers specified by --requestheader-username-headers.|Generally do not depend on authorization being already done for incoming requests. Always provide the root certificate bundle. This allows authorization to happen on each incoming request through `requestheader-allowed-names`.|
+|`tls-cipher-suites strings`|Comma-separated list of cipher suites for the server. If omitted, the default Go cipher suites will be used.|Always provide a list of preferred cipher suites. This ensures encryption never happens with insecure cipher suites.|
+{{</table>}}
+
+## Scheduling configurations
+The cluster administrator needs be careful with the plugins that use the `queueSort`, `filter`, or permit [extension points](https://github.com/docs/reference/scheduling/config/#extension-points).
+Scheduling happens in a series of stages that are exposed through the extension points.
+As a cluster administrator, you can enable plugins that define their own extension points.
+Doing so can affect the defined scheduling behaviors of the kube-scheduler in your cluster.
+
+### Extension points
+
+Exactly one plugin that uses the `queueSort` extension point can be enabled at a time. Any plugins that use `queueSort` should be scrutinized.
+
+Plugins that implement the `filter` extension point can potentially mark all nodes as unschedulable. This can bring scheduling of new pods to a halt.
+
+Plugins that implement the `permit` extension point  can prevent or delay the binding of a Pod. Such plugins should be thoroughly reviewed by the cluster administrator.
+
+When using a plugin that is not one of the [default plugins](https://kubernetes.io/docs/reference/scheduling/config/#scheduling-plugins), consider disabling the `queueSort`, `filter` and `permit` extension points as follows:
+
+```yaml
+apiVersion: kubescheduler.config.k8s.io/v1
+kind: KubeSchedulerConfiguration
+profiles:
+  - schedulerName: default-scheduler
+  - schedulerName: my-custom-scheduler
+    plugins:
+      queueSort:
+        disabled:
+        - name: '*'
+      filter:
+        disabled:
+        - name: '*'
+      permit:
+        disabled:
+        - name: '*'
+```
+This creates 2 scheduler profiles `default-scheduler` and ` my-custom-scheduler`. Whenever a podSpec does not have `.spec.schedulerName`, the kube-apiserver will use default plugin. The other option for pods would be to use `my-custom-scheduler` which has `queueSort`, `filter` and `permit` extension points disabled.
```

Comment:
@raesene @sftim, does this look good for this hardening guide?

Reply:
1. Considering our style guide, I recommend:
   ```suggestion
   This creates two scheduler profiles: `default-scheduler` and ` my-custom-scheduler`.
   Whenever the `.spec` of a Pod does not have a value for `.spec.schedulerName`,
   the kube-scheduler runs for that Pod, using its main configuration, and default plugin.
   If you define a Pod with `.spec.schedulerName` set to `my-custom-scheduler`, the
   kube-scheduler also runs but with a custom configuration; in that custom configuration,
   the  `queueSort`, `filter` and `permit` extension points are disabled.
   
   If you use this kube-scheduler configuration, and don't run any custom scheduler, and
   you then define a Pod with  `.spec.schedulerName` set to `nonexistent-scheduler` (or
   any other scheduler name that doesn't exist in your cluster), then the Pod stays pending
   scheduling. You would see a `FailedScheduling` event for the Pod and could manually
   remove that Pod using `kubectl delete`.
   ```
2. I am not convinced this is the right page for that advice. I think this belongs in a page about scheduling, and the hardening guide should hyperlink there.

Reply:
I added the this configuration here to explain how to disable `queueSort`, `filter` and `permit` extensions based on the previous feedback. We can move it to scheduler configuration but this specifically seems liked a security hardening direction. If we were to move this to a page about scheduling then it would need to cover scheduling guidelines with even more details.

Reply:
In this section I'm trying to explain how to harden scheduler scheduler security when working with custom schedulers and extension points.

Reply:
OK. So, I suggest the heading:
```dff
-### Extension points
### Scheduler security for custom schedulers
```

Does this cover:
- your own scheduler code
- a custom scheduler using a mix of Kubernetes code and your own code
- plain kube-scheduler, customized only through configuration files

Folks reading this need to be able to work out which of those scenarios we might (might be all, might be a subset). I can't tell, so I think other readers will also not be certain.

Reply:
Oh I see what you mean. I'll update this section to be more descriptive 👍